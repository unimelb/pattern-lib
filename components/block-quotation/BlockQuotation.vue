<template>
  <blockquote
    :class="{ 'block-quotation--bdr-top': borderTop, 'block-quotation--bdr-btm': borderBottom , 'block-quotation--small': small }"
    class="block-quotation">
    <p
      :class="{ 'block-quotation__content--large-quotes': largeQuotes }"
      class="block-quotation__content">
      <slot />
    </p>
    <cite
      v-if="author"
      class="block-quotation__author">{{ author }}
      <div
        v-if="subCite"
        class="block-quotation__sub-cite">{{ subCite }}</div>
    </cite>
  </blockquote>
</template>

<script>
export default {
  props: {
    author: {
      type: String,
      default: '',
    },
    subCite: {
      type: String,
      default: '',
    },
    borderTop: {
      type: Boolean,
      default: false,
    },
    borderBottom: {
      type: Boolean,
      default: false,
    },
    largeQuotes: {
      type: Boolean,
      default: false,
    },
    small: {
      type: Boolean,
      default: false,
    },
  },
};
</script>
