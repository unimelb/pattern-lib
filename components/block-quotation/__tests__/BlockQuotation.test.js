import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import BlockQuotation from '../BlockQuotation.vue';

expect.extend(toHaveNoViolations);

describe('BlockQuotation', () => {
  it('should match snapshot', () => {
    const result = shallow(BlockQuotation).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(BlockQuotation).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
