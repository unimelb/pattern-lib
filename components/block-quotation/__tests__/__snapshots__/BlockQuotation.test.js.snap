// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockQuotation should match snapshot 1`] = `
<blockquote
  class="block-quotation"
>
  <p
    class="block-quotation__content"
  />
   
  <!---->
</blockquote>
`;
